docs/examples/201/log-analytics-with-solutions-and-diagnostics/main.bicep

```diff
@@ -0,0 +1,124 @@
+param location string {
+    default: resourceGroup().location
+    allowed: [
+        'australiacentral' 
+        'australiaeast' 
+        'australiasoutheast' 
+        'brazilsouth'
+        'canadacentral' 
+        'centralindia' 
+        'centralus' 
+        'eastasia' 
+        'eastus' 
+        'eastus2' 
+        'francecentral' 
+        'japaneast' 
+        'koreacentral' 
+        'northcentralus' 
+        'northeurope' 
+        'southafricanorth' 
+        'southcentralus' 
+        'southeastasia'
+        'switzerlandnorth'
+        'switzerlandwest'
+        'uksouth' 
+        'ukwest' 
+        'westcentralus' 
+        'westeurope' 
+        'westus' 
+        'westus2' 
+    ]
+}
+
+param logAnalyticsWorkspaceName string = 'la-${uniqueString(resourceGroup().id)}'
+
+var vmInsights = {
+    name: 'VMInsights(${logAnalyticsWorkspaceName})'

```

Comment:
Did you intend to have the parens `()` as part of the string here? In other words, the parens are going to be in the final calculated name and will show up in the resource names

Reply:
I did intend to have the parens `()` as part of the string. In referencing the documentation on enabling Azure Monitor for VMs and Azure Monitor for Containers, that syntax is used. You can view the sample ARM Template at this link: [https://docs.microsoft.com/en-us/azure/azure-monitor/insights/vminsights-enable-resource-manager](url)

I tried using `name: 'VMInsights-${logAnalyticsWorkspaceName}'` as well as some other options, and the deployment errored out.

Reply:
Ah, ok. Looks good to me then.